--- a/arch/mips/bcm47xx/prom.c
+++ b/arch/mips/bcm47xx/prom.c
@@ -165,13 +165,17 @@ void __init bcm47xx_prom_highmem_init(void)
 
 	early_tlb_init();
 
-	if (!extmem)
-		return;
-
-	pr_warn("Found %lu MiB of extra memory, but highmem is unsupported yet!\n",
-		extmem >> 20);
-
-	/* TODO: Register extra memory */
+	if (!extmem) {
+		pr_warn("Found 0 MiB of extra memory\n");
+	}
+	/* Create permanent uncached TLB mapping for highmem */
+	add_permanent_tlb_entry(ENTRYLO(0x80000000),
+				ENTRYLO(0x80000000 + (256 << 20)),
+				EXTVBASE, PM_256M);
+
+	/* Register the highmem region */
+	pr_info("Found %lu MiB of highmem at 0x%08lx\n",
+	memblock_add(0x80000000 + lowmem, extmem));
 }
 
 #endif /* defined(CONFIG_BCM47XX_BCMA) && defined(CONFIG_HIGHMEM) */
